Validate hex config keys

theodowling opened the pull request:
This is to fix issue https://github.com/hexpm/hex/issues/617

Please let me know if I need to make any adjustments.

ericmj commented:
This looks great. Please update the failing test also to include an `assert_raise Mix.Error, "message", fn -> ... end` to check for the invalid key.

theodowling commented:
Great! Thanks for the tip 👍 
I'll have a look at fixing the failing tests when I get a moment tonight.

theodowling commented:
I've fixed the failing tests and added another test for invalid keys as requested. 
But I'm not sure why one of the build's failed 🤔
Can you maybe have a look?
